src/Datadog.Trace/Tracer.cs

```diff
@@ -122,6 +123,23 @@ internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler samp
             {
                 InitializeLibLogScopeEventSubscriber(_scopeManager);
             }
+
+#if NETSTANDARD2_0
+            // for now, only support DiagnosticSource for ASP.NET Core and only when running on .NET Core
+            if (Settings.DiagnosticSourceEnabled &&
+                Settings.IsIntegrationEnabled(AspNetCoreDiagnosticObserver.IntegrationName))
+            {
+                // instead of adding a hard dependency on DiagnosticSource,
+                // check that it is available before trying to use it
+                var type = Type.GetType("System.Diagnostics.DiagnosticSource, System.Diagnostics.DiagnosticSource", throwOnError: false);
+
+                if (type != null)
+                {
+                    DiagnosticManager = InitializeDiagnosticObservers();
+                    DiagnosticManager?.Start();
```

Comment:
nit: Can we do without the `?.` operator here? This should always be non-null, right?